service/history/workflow/workflow_task_state_machine.go

```diff
@@ -133,6 +136,11 @@ func (m *workflowTaskStateMachine) ReplicateTransientWorkflowTaskScheduled() (*W
 		Attempt:             m.ms.GetExecutionInfo().WorkflowTaskAttempt,
 		ScheduledTime:       timestamp.TimePtr(m.ms.timeSource.Now()),
 		StartedTime:         timestamp.UnixOrZeroTimePtr(0),
+		// QUESTION: should we preserve these here? this is used by mutable state rebuilder. it
+		// seems like the same logic as case 1 above applies: if a failover happens right after
+		// this, then AddWorkflowTaskStartedEvent will rewrite these anyway. is that correct?
```

Comment:
Yeah I think the value here doesn't matter as it will get overwritten anyway, either when starting the workflow task (if failover happens) or when replicating started event (no failover).

I am not familiar with the high level design of this feature so I probably overlooked something. Why those two fields need to be persisted in mutable state in addition to the workflow task started event?